File: app/src/main/java/com/cyl/musiclake/ui/settings/SettingsActivity.java
package com.cyl.musiclake.ui.settings;


import android.content.Intent;
import android.view.MenuItem;

import com.cyl.musiclake.R;
import com.cyl.musiclake.ui.base.BaseActivity;

public class SettingsActivity extends BaseActivity {


    @Override
    protected int getLayoutResID() {
        return R.layout.activity_setting;
    }

    @Override
    protected void initView() {
    }

    @Override
    protected void initData() {
        android.app.FragmentManager fragmentManager = getFragmentManager();
        fragmentManager.beginTransaction()
                .replace(R.id.container, SettingsFragment.newInstance()).commit();
    }

    @Override
    protected void initInjector() {

    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        int id = item.getItemId();
        if (id == android.R.id.home) {
            finish();
            return true;
        }
        return super.onOptionsItemSelected(item);
    }

    @Override
    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
        super.onActivityResult(requestCode, resultCode, data);
    }

}

File: app/src/main/java/com/cyl/musiclake/ui/settings/SettingsActivity.kt
package com.cyl.musiclake.ui.settings

import android.content.Intent
import android.view.MenuItem
import com.cyl.musiclake.R
import com.cyl.musiclake.ui.base.BaseActivity
import com.cyl.musiclake.ui.base.BaseContract

class SettingsActivity : BaseActivity<BaseContract.BasePresenter<*>>() {
    override fun getLayoutResID(): Int {
        return R.layout.activity_setting
    }

    override fun initView() {}
    override fun initData() {
        supportFragmentManager
                .beginTransaction()
                .replace(R.id.container, SettingsFragment())
                .commit()
    }

    override fun initInjector() {}
    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        val id = item.itemId
        if (id == android.R.id.home) {
            finish()
            return true
        }
        return super.onOptionsItemSelected(item)
    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        super.onActivityResult(requestCode, resultCode, data)
    }
}